.github/workflows/my-cd-pipeline.yml

```diff
@@ -0,0 +1,30 @@
+name: Continuous Deployment
+on:
+  push:
+    branches:
+      - main
```

Comment:
Because this is branch restricted, _if we're careful_ then we should be confident that the tests have run before this is deployed, but we don't get any guarantees. For example, if this was pushed & review, but then a merge conflict arose and you dealt with it when merging, we wouldn't necessarily have rerun the tests before this deployment occurs. The typical solution with GitHub Actions is to define both jobs in the same yaml file, and specify that one [needs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds) the other